apps/fz_http/priv/repo/migrations/20221201216913_create_gateway_firewall_policies.exs

```diff
@@ -0,0 +1,62 @@
+defmodule FzHttp.Repo.Migrations.CreateGatewayFirewallPolicies do
+  use Ecto.Migration
+
+  @create_query "CREATE TYPE protocol_enum AS ENUM ('tcp', 'udp')"
+  @drop_query "DROP TYPE protocol_enum"
+
+  def change do
+    execute(@create_query, @drop_query)
+
+    create table(:gateway_firewall_policies, primary_key: false) do
+      add(:id, :uuid, primary_key: true, default: fragment("gen_random_uuid()"))
+      add(:gateway_id, references(:gateways, on_delete: :delete_all, type: :uuid), null: false)
+      add(:user_id, references(:users, column: :uuid, type: :uuid))
+      add(:destination, :inet, null: false)
+      add(:port_range, :int4range)
+      add(:protocol, :protocol_enum)
```

Comment:
Maybe we can create a constraint `check :protocol not null and :port_range not null or :protocol null` so that we don't have to check that constraint in the ecto schema.

Reply:
Good idea, I accidentally put in my older constraint I had just checking if a range had a protocol, but just pushed up the correct constraint you described here.

Reply:
I also added another and to the constraint to make sure port range is not null when protocol is null.